src/juju/actions.js

```diff
@@ -2,74 +2,87 @@ import { fetchAndStoreModelStatus } from "juju";
 
 // Action labels
 export const actionsList = {
+  clearModelData: "CLEAR_MODEL_DATA",
   fetchModelList: "FETCH_MODEL_LIST",
   updateModelInfo: "UPDATE_MODEL_INFO",
   updateModelStatus: "UPDATE_MODEL_STATUS",
   updateModelList: "UPDATE_MODEL_LIST"
 };
 
 // Action creators
+
+export function clearModelData() {
+  return {
+    type: actionsList.clearModelData
+  };
+}
+
+// Thunks
+
 /**
   @param {Array} models The list of models to store.
-  @returns {Object} An action for Redux.
 */
 export function updateModelList(models) {
-  return {
-    type: actionsList.updateModelList,
-    payload: models
+  return function updateModelList(dispatch, getState) {
+    dispatch({
+      type: actionsList.updateModelList,
+      payload: models
+    });
   };
 }
 
 /**
   @param {String} modelUUID The modelUUID of the model to store the
     status under.
   @param {Object} status The status data as returned from the API.
-  @returns {Object} An action for Redux.
  */
 export function updateModelStatus(modelUUID, status) {
-  return {
-    type: actionsList.updateModelStatus,
-    payload: {
-      modelUUID,
-      status
-    }
+  return function updateModelStatus(dispatch, getState) {
+    dispatch({
+      type: actionsList.updateModelStatus,
+      payload: {
+        modelUUID,
+        status
+      }
+    });
   };
 }
 
 /**
   @param {Object} modelInfo The model info data as returned from the API.
-  @returns {Object} An action for Redux.
  */
 export function updateModelInfo(modelInfo) {
-  return {
-    type: actionsList.updateModelInfo,
-    payload: modelInfo
+  return function updateModelInfo(dispatch, getState) {
+    dispatch({
+      type: actionsList.updateModelInfo,
+      payload: modelInfo
+    });
   };
 }
 
-// Thunks
-
 /**
-  Fetches the model list from the supplied Juju controller.
+  Fetches the model list from the supplied Juju controller. Requires that the
+  user is logged in to dispatch the retrieved data from listModels.
   @returns {Object} models The list of model objects under the key `userModels`.
 */
 export function fetchModelList() {
-  return async function thunk(dispatch, getState) {
-    const conn = getState().root.controllerConnection;
+  return async function fetchModelList(dispatch, getState) {
+    const state = getState();
+    const conn = state.root.controllerConnection;
     const modelManager = conn.facades.modelManager;
     const models = await modelManager.listModels({ tag: conn.info.identity });
     dispatch(updateModelList(models));
   };
 }
 
 export function fetchModelStatus(modelUUID) {
-  return async function thunk(dispatch, getState) {
+  return async function fetchModelStatus(dispatch, getState) {
     const jujuState = getState().juju;
     if (jujuState.modelStatuses && jujuState.modelStatuses[modelUUID]) {
       // It already exists, don't do anything as it'll be updated shortly
       // by the polling loop.
```

Comment:
Heh yeah the naming of this function is a bit confusing, as no matter what it never returns the data, we use a uni-directional data flow so this will update the data in the database instead. Doing this makes it idempotent as we continuously fetch the status elsewhere.